transcendence-app/src/socket/socket.gateway.ts

```diff
@@ -39,39 +39,37 @@ export class SocketGateway
     if (!user) {
       client.disconnect();
     } else {
-      this.connectedUsers.add(user.id);
+      this.onlineUserIds.add(user.id);
       const sessionId = client.request.session.id;
       // Join the session ID room to keep track of all the clients linked to this session ID
       client.join(sessionId);
       // Join the user ID room to keep track of all the connected clients
       // (one user could connect from multiple private tabs or browsers with different session IDs)
       client.join(user.id);
-      const matchingSockets: Set<SocketId> = await this.server
-        .in(user.id)
-        .allSockets();
-      const isConnectedOnce = matchingSockets.size === 1;
+      const matchingSockets: RemoteSocket<DefaultEventsMap, any>[] =
+        await this.server.in(user.id).fetchSockets();
```

Comment:
Changed this to `fetchSockets()` because the `allSockets()` method is deprecated in the next major socketIO version.